File: use-cases/customer-growth-job-service/e2e/deploy/python-sqlalchemy-service/app/authorization/customer_growth_job_resource_mapping.py
from authguard_adapter.model import PathMap, ResourceSqlMapping, SegmentMap

ACTION_CREATE = "customer-growth.job.create"
ACTION_READ = "customer-growth.job.read"
ACTION_UPDATE = "customer-growth.job.update"
ACTION_DELETE = "customer-growth.job.delete"


def customer_growth_job_mapping() -> ResourceSqlMapping:
    return ResourceSqlMapping(
        SegmentMap.constant("prod"),
        SegmentMap.constant("customer-growth"),
        SegmentMap.column("region"),
        SegmentMap.column("tenant_id"),
        (
            PathMap.literal("workspace"),
            PathMap.column("workspace_id"),
            PathMap.literal("project"),
            PathMap.column("project_id"),
            PathMap.literal("job"),
            PathMap.column("job_id"),
        ),
    )

File: use-cases/customer-growth-job-service/e2e/deploy/golang-sqlx-service/pkg/authorization/customer_growth_job_resource_mapping.go
package authorization

import "authguard/adapters/golang/model"

const (
	ActionCreate = "customer-growth.job.create"
	ActionRead   = "customer-growth.job.read"
	ActionUpdate = "customer-growth.job.update"
	ActionDelete = "customer-growth.job.delete"
)

func CustomerGrowthJobResourceMapping() model.ResourceSQLMapping {
	return model.ResourceSQLMapping{
		Partition: model.ConstSegment("prod"),
		Service:   model.ConstSegment("customer-growth"),
		Region:    model.ColumnSegment("region"),
		Tenant:    model.ColumnSegment("tenant_id"),
		Path: []model.PathMap{
			model.LiteralPath("workspace"),
			model.ColumnPath("workspace_id"),
			model.LiteralPath("project"),
			model.ColumnPath("project_id"),
			model.LiteralPath("job"),
			model.ColumnPath("job_id"),
		},
	}
}
